scripts/ci/github/check_bootnodes.sh

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+
+# In this script, we check each bootnode for a given chainspec file and ensure they are contactable.
+# We do this by removing every bootnode from the chainspec with the exception of the one
+# we want to check. Then we spin up a node using this new chainspec, wait a little while
+# and then check our local node's RPC endpoint for the number of peers. If the node hasn't
+# been able to contact any other nodes, we can reason that the bootnode we used is not well-connected
+# or is otherwise uncontactable.
+
+# shellcheck source=scripts/ci/common/lib.sh
+source "$(dirname "${0}")/../common/lib.sh"
+CHAINSPEC_FILE="$1"
+RUNTIME=$(basename "$CHAINSPEC_FILE" | cut -d '.' -f 1)
+
+trap cleanup EXIT INT TERM
+
+cleanup(){
+    echo "[+] Script interrupted or ended. Cleaning up..."
+    # Kill all the polkadot processes
+    killall polkadot > /dev/null 2>&1
+    exit $1
+}
+
+# count the number of bootnodes
+BOOTNODES=$( jq -r '.bootNodes | length' "$CHAINSPEC_FILE" )
+# Make a temporary dir for chainspec files
+# Store an array of the bad bootnodes
+BAD_BOOTNODES=()
+GOOD_BOOTNODES=()
+PIDS=()
+
+echo "[+] Checking $BOOTNODES bootnodes for $RUNTIME"
+for i in $(seq 0 $((BOOTNODES-1))); do
+    BOOTNODE=$( jq -r .bootNodes["$i"] < "$CHAINSPEC_FILE" )
+    # Check each bootnode in parallel
+    check_bootnode "$BOOTNODE" "$CHAINSPEC_FILE" &
+    PIDS+=($!)
+    # Hold off 10 seconds between attempting to spawn nodes to stop the machine from getting overloaded
```

Comment:
This does not really prevent it from overloading, or?

Reply:
Initially it was a 1-second holdoff between spawning but it led to the test being flaky when running on the github runners (worked mostly fine locally so long as I didn't need to do anything else on this machine), which I determined to be a result of heavy initial load of launching the executable. Could be avoided by using a beefier runner but the test is not that long-running so it seemed like a reasonable tradeoff

Reply:
Also considered that. Maybe we can revisit that in the future if we end up with around 100 bootnodes or more per runtime, but felt like a bit of premature optimisation at the moment ;)